fastapi_security/api.py

```diff
@@ -37,6 +43,9 @@ def __init__(self, *, user_permission_class: Type[UserPermission] = UserPermissi
     def init_basic_auth(self, basic_auth_credentials: IterableOfHTTPBasicCredentials):
         self.basic_auth.init(basic_auth_credentials)
 
+    def init_basic_auth_with_digest(self, salt: str, basic_auth_with_digest_credentials: IterableOfHTTPBasicCredentialsDigest):
+        self.basic_auth_with_digest.init(salt, basic_auth_with_digest_credentials)
+
```

Comment:
Previous comment would mean that this method won't be needed. The `salt` argument would be passed through to `BasicAuthWithDigestValidator.init` in **kwargs of `init_basic_auth`.

```suggestion
    def init_basic_auth(self, basic_auth_credentials: IterableOfHTTPBasicCredentials, **kwargs):
        self.basic_auth.init(basic_auth_credentials, **kwargs)
```

Reply:
I had considered this before, and I didn't quite like it. Suppose, there are two ways of configuring the basic auth (with the class param):
```python
from fastapi_security import FastAPISecurity
security = FastAPISecurity()
security.init_basic_auth(basic_credentials)
```
and
```python
from fastapi_security import BasicAuthWithDigestValidator, FastAPISecurity
security = FastAPISecurity(basic_auth_class=BasicAuthWithDigestValidator)
security.init_basic_auth(basic_credentials, salt=salt)
```

The issues that I noticed here:
- the unsafe way is sort of incentivised, because digest-based basic auth now requires an extra import and it is even more convenient to without it
- `basic_auth_class=fastapi_security.BasicAuthWithDigestValidator` and `salt=salt` params are tightly coupled: one cannot be used with the other one, and if one of the two digest configuration steps are missed, `init_basic_auth` method will fail with an obscure `TypeError` error about either an unexpected or a missing required `salt` kwarg

What if we do smth like this:

```python
class FastAPISecurity:
     def __init__(self, ...):
        ...
        # empty validator, for now its use is reduced to simply returning `False` for `basic_auth.is_configured()`
        # if `init_basic_auth`/`init_basic_auth_with_digest` is never called
        self.basic_auth = BasicAuthValidator()  

    def init_basic_auth(self, credentials):
        self.basic_auth = BasicAuthValidator()
        self.basic_auth.init(credentials)

    def init_basic_auth_with_digest(self, credentials, *, salt):
        self.basic_auth = BasicAuthWithDigestValidator()
        self.basic_auth.init(credentials, salt=salt)
```

Reply:
Much better, thanks.